src/Features/CSharp/Portable/Copilot/CSharpImplementNotImplementedExceptionDiagnosticAnalyzer.cs

```diff
@@ -29,29 +32,138 @@ protected override void InitializeWorker(AnalysisContext context)
         {
             var notImplementedExceptionType = context.Compilation.GetTypeByMetadataName(typeof(NotImplementedException).FullName!);
             if (notImplementedExceptionType != null)
-                context.RegisterOperationAction(context => AnalyzeThrow(context, notImplementedExceptionType), OperationKind.Throw);
+            {
+                context.RegisterOperationBlockAction(context =>
+                {
+                    using var _ = SharedPools.Default<ConcurrentSet<Location>>().GetPooledObject(out var reportedLocations);
```

Comment:
you don't need a concurrent set in *this* case.  you registerd a block action (not a start action).  so the block action will call (concurrently) on each block.  but you're not sharing data across them.

Reply:
switched to hashset, because otherwise we'd get diagnostic reported on locations below twice, because the location of the owning entity would be identical to the location where the exception is thrown:

```csharp
using System;
 
class C
{
    int P => {|IDE3000:throw new NotImplementedException()|};
}
```
 
contrast that with

```csharp
using System;
 
class C
{
    public int P
    {
        {| IDE3000:get |} { {| IDE3000:throw new NotImplementedException();|} }
        {| IDE3000:set |} { {| IDE3000:throw new NotImplementedException();|} }
    }
}
```